sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -146,7 +151,25 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
           predicateReference ++ condition.references,
           hasHitFilter = true,
           hasHitSelectiveFilter = hasHitSelectiveFilter || isLikelySelective(condition))
-      case _: LeafNode => hasHitSelectiveFilter
+      case ExtractEquiJoinKeys(joinType, _, _, _, _, left, right, hint) =>
+        // Runtime filters use one side of the [[Join]] to build a set of join key values and prune
+        // the other side of the [[Join]]. It's also OK to use a superset of the join key values to
+        // do the pruning. For inner [[Join]]s, one side of the [[Join]] always produces a superset
```

Comment:
We can remove `For inner [[Join]]s ...` as the actual code allows more join types.